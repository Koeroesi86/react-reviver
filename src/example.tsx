/* eslint-disable react/prop-types */
import React, { FC } from "react";
import ReactDOM from "react-dom";
import { ReviverProvider, ReviverLayout} from "./index";
import { ComponentRegistry, RevivableComponentType } from "./types";

const Wrapper: FC<{ background: string }> = ({ children, background }) => (
  <div style={{ backgroundColor: background }}>{children}</div>
);
const Hello: FC<{ text: string }> = ({ text }) => <h1>{text}</h1>;

export const components: ComponentRegistry = {
  wrapper: Wrapper,
  hello: Hello,
}

type RevivableComponent =
  | RevivableComponentType<"wrapper", { background: string }, RevivableComponent>
  | RevivableComponentType<"hello", { text: string }, RevivableComponent>;

export const layout: RevivableComponent = {
  type: "wrapper",
  props: { background: "#efefef" },
  children: [
    { type: "hello", props: { text: "It works!" }, children: [] },
  ]
};

if (window && document.getElementById("root")) {
  ReactDOM.render(
    <ReviverProvider components={components}>
      Example:
      <ReviverLayout data={layout}/>
    </ReviverProvider>,
    document.getElementById("root"),
  );
}