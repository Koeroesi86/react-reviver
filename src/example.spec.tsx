import { render } from "@testing-library/react";
import React from "react";
import { ReviverLayout, ReviverProvider } from "./index";
import { components, layout } from "./example";

describe("example render", () => {
  it("should render", () => {
    const result = render(
      <ReviverProvider components={components}>
        Example:
        <ReviverLayout data={layout}/>
      </ReviverProvider>
    );

    expect(result.asFragment()).toMatchSnapshot();
  });
})