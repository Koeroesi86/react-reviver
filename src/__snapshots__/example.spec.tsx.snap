// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example render should render 1`] = `
<DocumentFragment>
  Example:
  <div
    style="background-color: rgb(239, 239, 239);"
  >
    <h1>
      It works!
    </h1>
  </div>
</DocumentFragment>
`;
